lib/Service/AutoProvisioningService.php

```diff
@@ -80,6 +80,17 @@ public function createUser($userInfo): IUser {
 		if (!$emailOrUserId) {
 			throw new LoginException("Configured attribute $attribute is not known.");
 		}
+
+		$openIdConfig = $this->getOpenIdConfiguration();
+		$provisioningClaim = $openIdConfig['auto-provision']['provisioning-claim'] ?? null;
+		if ($provisioningClaim) {
+			$this->logger->debug('ProvisioningClaim is defined for auto-provision');
+			$provisioningAttribute = $openIdConfig['auto-provision']['provisioning-attribute'] ?? null;
+			if (! \in_array($provisioningAttribute, $userInfo->$provisioningClaim, true)) {
```

Comment:
$userInfo->$provisioningClaim could blow up in case it is not set

Reply:
When this claim is not defined, you hit the error; `Required provisioning attribute eduperson_entitlement is not set.`
But can add an extra check of this claim is defined.